CONTRIBUTING.md

```diff
@@ -1,3 +1,101 @@
 # Contributing Guidelines
 
-TODO.
+Thanks for your interest in contributing to the Creative Labs website, aka Sunshine! ❤️
+
+These guidelines describe how to set up your environment and submit your own contributions! Please read it 
+carefully and, if there is anything that isn't up-to-date, feel free to submit a PR with your corrections 🙃
+
+- [Getting Started](#Getting-Started)
+- [Pull Requests](#Pull-Requests)
+  - [Step 1: Open an Issue](#Step-1:-Open-an-Issue)
+  - [Step 2: Design](#Step-2:-Design)
+  - [Step 3: Beep Boop](#Step-3:-Beep-Boop)
+  - [Step 4: Commit](#Step-4:-Commit)
+  - [Step 5: Pull Request](#Step-5:-Pull-Request)
+  - [Step 6: Merge](#Step-6:-Merge)
+
+## Getting Started
+
+We use [`yarn`](https://classic.yarnpkg.com/en/docs/install#mac-stable) as our package manager.
+
+The basic commands to get this repository and start are:
+
+```
+$ git clone https://github.com/UCLA-Creative-Labs/sunshine.git
+$ cd sunshine
+$ yarn install
+```
+
+To expedite local development, run locally:
+
+```
+$ yarn install
+```
+
+To test the production build, run:
+
+```
+$ yarn build
+$ cd dist
+$ python3 -m http.server
+```
+
+## Pull Requests
+
+The following should be the general workflow of your pull requests. We have protected `master` so no one can 
+push directly to master. We also use [`mergify`](https://mergify.io/) to automate our merging process. No one 
+should be merging directly to `master`. Check out our [`.mergify.yml`](.mergify.yml) file for our conditions.
+
+### Step 1: Open an Issue
+
+If there isn't an existing issue, open an issue describing what you intend to contribute. Communicating visual and 
+code design proactively leads to less overhead for everyone and allows for collaboration. 
+
+### Step 2: Design
+
+Always reference a visual design when developing and make sure to link that design in the discussion. In terms of 
+code design, make sure to read through the [design guidelines](DESIGN_GUIDELINES.md) for general patterns we follow. 
+
+### Step 3: Beep Boop
+
+Beep boop. Beep boop. Here are some general guidelines:
+
+- In general, follow the style of code around you
+- Coding Style (Abridged)
+  - 2 space indentation
+  - 120 characters width
+  - files should be lowercased and kebab cased (i.e. foo-bar.ts)
+  - objects and variables should be camel cased (i.e. fooBar )
```

Comment:
Do you wanna add that class names should be PascalCase and functions should also be camelCase?

Reply:
so files, classes and variables should have a different type of casing imo..

I can understand the idea that file and classes should be both PascalCased since most files will export a singular class..

yeah in this case we can do PascalCase for files + classes and functions should be camelCase as long as we distinct that classes be React components